app/services/reports/facility_progress_service.rb

```diff
@@ -32,6 +46,19 @@ def dimension_combinations_for(indicator)
 
     private
 
+    attr_reader :diabetes_enabled
+
+    memoize def daily_registrations_grouped_by_day
+      diagnosis = diabetes_enabled ? :all : :hypertension
+      RegisteredPatientsQuery.new.count_daily(facility, diagnosis: diagnosis, last: 30)
+    end
+
+    memoize def daily_follow_ups_grouped_by_day
+      scope = Reports::DailyFollowUp.with_hypertension
+      scope = scope.or(Reports::DailyFollowUp.with_diabetes) if diabetes_enabled
+      scope.where(facility: facility).group(:day_of_year).count
+    end
```

Comment:
These should get moved into the repository, but that will be fine for a subsequent PR.